### PR TITLE
Warn when JAX-RS @Path is used along with MP Health annotations

### LINKED ISSUE #10720: @Liveness probe doesn't work with @Path
Quarkus 1.6.0.FINAL

the health probe annoted with @Liveness and @Path   won't be returned when we call http://localhost:8080/health/

I took the example : quarkus-quickstarts\microprofile-health-quickstart 

and when I do

mvn quarkus:dev
http://localhost:8080/health/

the output is : 

```
{
    "status": "UP",
    "checks": [
        {
            "name": "Simple health check",
            "status": "UP"
        },
        {
            "name": "Health check with data",
            "status": "UP",
            "data": {
                "bar": "barValue",
                "foo": "fooValue"
            }
        },
        {
            "name": "Database connection health check",
            "status": "UP"
        }
    ]
}
```

but if I add @Path in a class : SimpleHealthCheck that have @Liveness, the output became

```

{
    "status": "UP",
    "checks": [
        {
            "name": "Health check with data",
            "status": "UP",
            "data": {
                "bar": "barValue",
                "foo": "fooValue"
            }
        },
        {
            "name": "Database connection health check",
            "status": "UP"
        }
    ]
}
```

### PR DESCRIPTION
**Pull request by geoand**
It's not unreasonable for users to expect these annotations to work
together, so let's be explicit that they don't.

Resolves: #10720

### CODE REVIEW

File: extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java [hunk 1]
```diff
@@ -176,6 +178,10 @@ public void defineHealthRoutes(BuildProducer<RouteBuildItem> routes,
             BeanArchiveIndexBuildItem beanArchiveIndex) {
         IndexView index = beanArchiveIndex.getIndex();
 
+        // log a warning if users try to use @Liveness or @Readiness with JAX-RS @Path
+        warnIfJaxRsPathUsed(index, LIVENESS);
+        warnIfJaxRsPathUsed(index, READINESS);
```
Comment (jmartisk) [inline extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java:183]: MP Health 3.0 will remove the deprecated `@Health` annotation, but while we still have it I think we should log a warning for that one too?

Comment (geoand) [reply extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java:183]: Sure thing

Comment (geoand) [reply extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java:183]: Done

**Review by jmartisk — APPROVED** [review]
LGTM!

**Pull request merged**